### PR TITLE
Upgrade embedded Kotlin to 1.8.20

### PR DESCRIPTION
**Pull request by eskatos**
* Fixes https://github.com/gradle/gradle/issues/24096
* Notes https://github.com/JetBrains/kotlin/releases/tag/v1.8.20
* What's new https://kotlinlang.org/docs/whatsnew1820.html

### CODE REVIEW

**Comment by eskatos** [comment]
@bot-gradle test APT

**Review by jbartok — APPROVED** [review]

**Review by ov7a — APPROVED** [review]

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Comment by eskatos** [comment]
@bot-gradle cancel

File: subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc [hunk 1]
```diff
@@ -154,6 +154,25 @@ The following method on `Configuration` is deprecated for removal:
...
 
+=== Potential breaking changes
+
+[[kotlin_1_8.20]]
+==== Upgrade to Kotlin 1.8.20
+
+The embedded Kotlin has been updated to link:https://github.com/JetBrains/kotlin/releases/tag/v1.8.20[Kotlin 1.8.20].
+For more information, see https://kotlinlang.org/docs/whatsnew1820.html[What's new in Kotlin 1.8.20].
+
+Note that there is a known issue with Kotlin compilation avoidance that can cause `OutOfMemory` exceptions in `compileKotlin` tasks if the compilation classpath contains very large JAR files.
```
Comment (ov7a) [inline subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc:165]: I would point explicitly that this applies to `buildSrc` and does not affect the application code. Better be safe and explicit here.

Comment (eskatos) [reply subprojects/docs/src/docs/userguide/migration/upgrading_version_8.adoc:165]: Good point. Added, PTAL

**Review by ov7a — APPROVED** [review]

**Comment by eskatos** [comment]
@bot-gradle test and merge

**Pull request merged**